scripts/fastlane/Fastfile

```diff
@@ -0,0 +1,157 @@
+
+    git_add(path: [gradle_properties_path, change_log_path])
+    git_commit(path: [gradle_properties_path, change_log_path], message: "chore(release): #{product_name} - #{next_version}")
+    push_to_git_remote(force: true)
+
+
+    current_branch = sh('git', 'branch', '--show-current')
+    repo_name_parts = options[:repo].split('/')
+    create_pull_request(base: 'main', 
+                        title: "chore(release): #{product_name} - #{next_version}", 
+                        body:pr_body, 
+                        repo: options[:repo], 
+                        api_token: ENV["RELEASE_MANAGER_TOKEN"])
+  end
+
+
+  desc "Checks out a tag and runs the uploadArchives gradle task."
+  lane :push_tag_to_maven do |options|
+    UI.message("Pushing release #{options[:release_tag]} to maven")
+    sh("git", "checkout", options[:release_tag])
+    publish_to_maven
+  end
+
+  desc "Update changelog"
+  private_lane :update_change_log do |options|
+    changelog_body = changelog_from_git_commits(
+      between: [options[:last_release_tag], "HEAD"],
+      merge_commit_filtering: "exclude_merges",
+      pretty: "- %s"
+    )
+
+    open(change_log_path, 'a') { |f|
+      f.puts "# Release Amplify Android #{options[:next_version]}\n\n"
+      f.puts "#{changelog_body}\n\n"
+    }
+  end
+
+  desc "Increment versions"
+  private_lane :update_gradle_properties do |options|
+    version = options[:version].to_s
+    segments = version.match(version_regex).captures # version.split('.')
+
+    UI.message("Updating versionName in gradle.properties")
+
+    set_key_value(file: gradle_properties_path, key: 'VERSION_NAME', value: version)
+
+  end
+
+  # Calculate the Android version code. Leaving this here for posterity
+  # since it appears it's not necessary to increment versionCode for Android libraries.
+  desc "Calculate a version code based on new version segments"
+  private_lane :calculate_version_code do |options|
+    segments = options[:segments]
+    prerelease = segments.length() > 3 ?  0 : segments[-1].to_i
+    # Android version code max version = 2_100_000_000
+
+    # Derive the version code from version segements and pre-release number (if applicable)
+    # With this approach, you're "limited" to 999 minor versions per major and 999 patches per minor version.
+    # Limits: Max version for 1.x would be 1.999.999 with version code 199_999_900
+    segments[0].to_i * 100_000_000 + segments[1].to_i * 100_000 + segments[2].to_i * 100 + prerelease
+  end
+
+  desc "Create the release commit."
+  private_lane :release_commit do |options|
+    next_version = options[:version]
+    commit_message = "chore: release #{next_version} [skip ci]"
+    sh('git', 'commit', '-am', commit_message)
+  end
+
+  desc "Tag in git and push to GitHub"
+  private_lane :add_tag do |options|
+    next_version = options[:version]
+
+    add_git_tag(tag: next_version)
+    push_git_tags(tag: next_version)
+  end
```

Comment:
Instead of just creating a tag, we should figure out how we can make a GitHub release, which should just create the tag automatically.  That can be added in a subsequent PR.

```suggestion
```

Reply:
Yup...this can go as well. I'll be working on creating the release in a separate workflow.